### PR TITLE
Add resource pack stuff

### PR DESCRIPTION
**Pull request by Blueyescat**
### Description
Adds an effect to send a resource pack to players, an event to listen to resource pack request responses, a condition to check the response in the event, an expression to get the last resource pack response of a player, a condition to check whether a player has a server resource pack loaded.

**Test Script**
```vb
#!send resource pack "https://drive.google.com/uc?export=download&id=1vvkWrR7LcGyveExwKchiSn6PAc3qVS-z" to player
on resource pack request response:
	send "%player's resource pack action%"
	if the resource pack was accepted:
		send "thanks"
	else if the resource pack was declined or failed to download:
		send "isn't loaded"
	if the resource pack wasn't accepted or success:
		send "^ the same"

on resource pack accept:
	send "Accepted"

on resource pack decline:
	send "Declined"

on resource pack success:
	send "Successfully Loaded"

on resource pack fail:
	send "Download Failed"

on resource pack decline or fail:
	send "Declined or failed"
```

---
**Target Minecraft Versions:** any
**Requirements:** Paper is needed for ExprLastResourcePackResponse and CondHasResourcePack only
**Related Issues:** None

### CODE REVIEW

File: src/main/java/ch/njol/skript/events/EvtResourcePackAction.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				.description("Called when a player takes action on a resource pack request sent via the ",
+						"<a href='effects.html#EffSendResourcePack'>send resource pack</a> effect. ",
+						"The <a href='expressions.html#ExprEventAction'>event action</a> expression can be used ",
+						"to get the resource pack action.",
+						"",
+						"This event will be triggered once when the player accepts or declines the resource pack request, ",
+						"and once when the resource pack is successfully installed or failed.")
+				.examples("on resource pack request action:",
+						"	if the action is decline or download fail:",
+						"",
```
Comment (Nicofisi) [inline src/main/java/ch/njol/skript/events/EvtResourcePackAction.java]: Might want to put something here maybe?

**Comment by Blueyescat** [comment]
@Nicofisi yeah `on resource pack something or something else:` works fine, already it is in the test script in the issue. Thanks 🙂

**Comment by Blueyescat** [comment]
As discussed in Discord, i used "state" and "response" instead of "action", removed the global event action expression and added a condition to check the response instead. I have updated the test script in the issue.

File: src/main/java/ch/njol/skript/conditions/CondResourcePack.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	
+	@Override
+	public boolean check(Event e) {
+		Status state = ((PlayerResourcePackStatusEvent) e).getStatus();
+		return states.check(e, state::equals, isNegated());
+	}
+	
+	@Override
+	public String toString(final @Nullable Event e, final boolean debug) {
+		return "resource pack was " + (isNegated() ? " not" : "") + states.toString(e, debug);
```
[the remark this answers is not in the document]
Comment (Blueyescat) [reply src/main/java/ch/njol/skript/conditions/CondResourcePack.java]: Good catch :+1:

**Pull request merged**